src/python/WMCore/MicroService/Unified/MSOutput.py

```diff
@@ -658,13 +679,29 @@ def docInfoUpdate(self, msOutDoc, pipeLine=None):
         """
 
         # Fill the destinationOutputMap first
-        if msOutDoc['isRelVal']:
-            destinationOutputMap = []
-            wflowDstSet = set()
-            updateDict = {}
-            for dataset in msOutDoc['OutputDatasets']:
-                _, dsn, procString, dataTier = dataset.split('/')
-                destination = set()
+        destinationOutputMap = []
+        wflowDstSet = set()
+        updateDict = {}
+        for dataset in msOutDoc['OutputDatasets']:
+            _, dsn, procString, dataTier = dataset.split('/')
+            # NOTE:
+            #    Data tiers that have been configured to be excluded will never
+            #    enter the destinationOutputMap
+            if dataTier in self.msConfig['excludeDataTier']:
+                # msg = "%s: %s: "
```

Comment:
I suggest to log the dataset name getting excluded and mention it's a data tier blacklisted.

Reply:
I was afraid of having thousands of those in a single run. But OK, I will log it.

Reply:
Ohh, if it is on the Producer thread, then yes, we will have hundreds of messages to the same workflow. With that in mind, perhaps we should then build the MSOutput document without any restrictions, and leave the possible "datatier filter" to be handled by the consumer thread.

Reply:
Yes, we can expose that thing into the Consumer though, but then we will deviate from the original idea - to have as little conditions as possible there, but to have everything predefined in the document instead. This was the concept I was relying on when I was consolidating the logic for both `RelVal`s and `NonRelval`s. Another reason why I put this in the Producer was that the Consumer does not iterate thorough the output datasets any more. It deals with the destinationOutputMap directly and builds the transfer requests based on the already aggregated dataset lists there.

Reply:
It's not an easy decision. There are pros and cons on both approaches. I think we can assume that these list of tiers to be skipped won't be around for more than 2 or 3 months, so it's something that we will soon be able to get rid of in the code.
I'd suggest to make a DEBUG level record them, just in case.